### PR TITLE
Remove Windows only sse

### PR DESCRIPTION
**Pull request by schultetwin1**
### Description of the changes:
- Enable SSE for depth to xyz code

### Before submitting a Pull Request:
- [x] I reviewed [CONTRIBUTING.md](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/CONTRIBUTING.md)
- [x] I [built my changes](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/building.md) locally
- [x] I ran the [unit tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md)
- [x] I ran the [functional tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md) with a device
- [ ] I ran the [performance tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md) with a device

### I tested changes on:
- [ ] Windows
- [x] Linux

### CODE REVIEW

File: src/transformation/rgbz.c [hunk 1]
```diff
@@ -791,38 +791,6 @@ k4a_buffer_result_t transformation_color_image_to_depth_camera_internal(
...
-            x = 0;
-            y = 0;
-            z = 0;
-        }
-
-        xyz_data_int16[3 * i + 0] = x;
-        xyz_data_int16[3 * i + 1] = y;
-        xyz_data_int16[3 * i + 2] = z;
-    }
-}
```
Comment (mbleyer) [inline src/transformation/rgbz.c]: I would rather not want to delete this code. It will be useful for people to understand what the SSE version is doing.

**Comment by schultetwin1** [comment]
@mbleyer this build break brings up a good error. On line 799 of rgbz.c we do the following cast:

```c
__m128 *x_table_m128 = (__m128 *)xy_tables->x_table;
```

`xy_tables->x_table` is a pointer to a float, which means it's pointing to a 4-byte aligned data member. We are casting that to a pointer to a `__m128`, meaning a pointer to a 16-byte aligned data member.

I've added assumptions in to get around this but is this something we should worry about?

**Comment by mbleyer** [comment]
> @mbleyer this build break brings up a good error. On line 799 of rgbz.c we do the following cast:
> __m128 *x_table_m128 = (__m128 *)xy_tables->x_table;
> xy_tables->x_table is a pointer to a float, which means it's pointing to a 4-byte aligned data member. We are casting that to a pointer to a __m128, meaning a pointer to a 16-byte aligned data member.
> I've added assumptions in to get around this but is this something we should worry about?

You are right. Maybe we should open a P3 bug on this.

File: src/transformation/rgbz.c (hunk 1 above)
Comment (schultetwin1) [reply src/transformation/rgbz.c]: Sounds good, put it back in

**Comment by schultetwin1** [comment]
> > @mbleyer this build break brings up a good error. On line 799 of rgbz.c we do the following cast:
> > __m128 *x_table_m128 = (__m128 *)xy_tables->x_table;
> > xy_tables->x_table is a pointer to a float, which means it's pointing to a 4-byte aligned data member. We are casting that to a pointer to a __m128, meaning a pointer to a 16-byte aligned data member.
> > I've added assumptions in to get around this but is this something we should worry about?
> 
> You are right. Maybe we should open a P3 bug on this.

Done, I've opened issue #318 and assigned it to you.

**Comment by Brent-A** [comment]
Does this change just ensure that the compiler is set to support SSE4.1? Don't we need to perform a runtime check as well to determine if the platform we are executing on supports those instructions, or we get a runtime failure?

**Comment by schultetwin1** [comment]
> Does this change just ensure that the compiler is set to support SSE4.1? Don't we need to perform a runtime check as well to determine if the platform we are executing on supports those instructions, or we get a runtime failure?

Good point, we should have a better failure than just a CPU fault on an unsupported instruction. Since this is already a problem (our depth engine already uses SSE4.1 instructions), I have created an issue to add that in a different PR. #324

**Comment by schultetwin1** [comment]
@mbleyer or @cdedmonds can you guys take a look at this today?

**Pull request merged**